code/modules/mob/dead/new_player/new_player.dm

```diff
@@ -161,6 +162,14 @@
 				to_chat(usr, "<span class='warning'>Server is full.</span>")
 				return
 
+		// Check if random role is requested
+		if(href_list["SelectedJob"] == "Random")
+			var/datum/job/job = SSjob.GetRandomJob(src)
+			if(!job)
+				to_chat(usr, "<span class='warning'>There is no randomly assignable Job at this time. Please manually choose one of the other possible options.</span>")
```

Comment:
Job should be jobs or job i cant grammar but it shouldn't be capitalized either case